### ISSUE TITLE
mock: introduce a match_no_header() negative matcher

### ISSUE THREAD

**Issue by lucab** [lipanski/mockito#12]
I'd like to see a new negative-matching helper added to Mock, which I guess may look like the following:
```
fn match_no_header(&mut self, field: &str, value: Option<&str>) -> &mut Self
```

The usecase is a test I want to write where I'm interested in ensuring that
a particular header has not been set on the request (optionally, only matching
the header key only).

**Comment by lipanski** [comment]
rogert that, sounds doable. I'll come back to you when I've got something.

**Comment by lipanski** [comment]
@lucabjust just released in 0.3.0 - check the details here: http://lipanski.github.io/mockito/generated/mockito/index.html#other-header-matchers

TL;DR:

```
use mockito::{mock, Matcher};

mock("GET", "/").match_header("content-type", Matcher::Missing).create();
```
